### DISCUSSION TITLE
Which of the following two approaches is recommended by Prisma?

### DISCUSSION THREAD

**Discussion by merrydance** [prisma/prisma#19458] [Q&A] [answered]
Which of the following two approaches is recommended by Prisma? 
```js
const update = await prisma.user.update({
  where: {
    id: 1,
  },
  data: {
    name: "Petunia",
    email: emailInput != null ? emailInput : undefined, // If null, don't include in update!
  },
});
```
and I have found that this one also works well：
```js
const { name,email } = request.body;
const update = await prisma.user.update({
  where: {
    id: 1,
  },
  data: {
    name,
    email，
  },
});
```

**Comment by ludralph** [comment, marked as answer]
Hi @merrydance 👋 

Both approaches you mentioned are valid and can be used with Prisma. Prisma does not explicitly recommend one approach over the other, as the choice depends on your specific use case. Approach 1 provides more control over which fields are included in the update operation, while Approach 2 is more concise and assumes that the fields will always be present in the request.

Choose the approach that best fits your needs. If you want to conditionally exclude fields from the update, Approach 1 would be a better choice. If you are confident that the fields will always be present and want a simpler syntax, Approach 2 is suitable.

**Reply by merrydance** [reply]
Thanks for your answer.
